source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/settings/workflow-definitions/common/form.tsx

```diff
@@ -0,0 +1,275 @@
+import * as React from 'react';
+import { compose } from 'recompose';
+import { Checkbox, Dropdown } from 'semantic-ui-react';
+import { withTemplateHelpers, Mut, Toggle } from 'react-action-decorators';
+
+import { withTranslations, i18nProps } from '@lib/i18n';
+import { isEmpty } from '@lib/collection';
+
+import * as toast from '@lib/toast';
+
+import { WorkflowDefinitionAttributes } from '@data/models/workflow-definition';
+import {
+  query,
+  buildOptions,
+  withLoader,
+  attributesFor,
+  StoreTypeResource,
+  WorkflowDefinitionResource
+} from '@data';
+
+
+interface IOwnProps {
+  workflowDefinition: WorkflowDefinitionResource;
+  storeType: StoreTypeResource;
+  storeTypes: StoreTypeResource[];
+  onSubmit: (attributes: WorkflowDefinitionAttributes, relationships: any) => void;
+  onCancel: () => void;
+}
+
+interface IState {
+  name?: string;
+  description?: string;
+  storeType?: StoreTypeResource;
+  storeTypeError?: string;
+  workflowScheme?: string;
+  workflowBusinessFlow?: string;
+  enabled?: boolean;
+}
+
+type IProps =
+  & i18nProps
+  & IOwnProps;
+
+
+@withTemplateHelpers
+class WorkflowDefinitionForm extends React.Component<IProps, IState> {
+  mut: Mut;
+  toggle: Toggle;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const { workflowDefinition, storeType } = props;
+
+
+    const {
+      name, description,
+      workflowScheme, workflowBusinessFlow,
+      enabled
+    } = attributesFor(workflowDefinition);
+
+
+    this.state = {
+      name: name || '',
+      description: description || '',
+      workflowScheme: workflowScheme || '',
+      workflowBusinessFlow: workflowBusinessFlow || '',
+      enabled: enabled || false,
+      storeType: storeType || null,
+      storeTypeError: ''
+    };
+    
+
+  }
+
+  isValidForm = () => {
+    const { storeType } = this.state;
+    const { t } = this.props;
+    const storeTypeError = isEmpty(storeType) ? t('admin.settings.workflowDefinitions.emptyStoreType') : '';
+    this.setState({storeTypeError});
+    return !isEmpty(storeType);
+  }
+
+  submit = async (e) => {
+    e.preventDefault();
+
+    const { onSubmit } = this.props;
+    const {
+      name, description, workflowScheme,
+      workflowBusinessFlow,
+      enabled, storeType
+    } = this.state;
+
+    if (this.isValidForm()) {
+      try {
+        await onSubmit({
+          name,
+          description,
+          workflowScheme,
+          workflowBusinessFlow,
+          enabled
+        },{
+          storeType
+        });
+      } catch(e) {
+        toast.error(e);
+      }
+    }
+  }
+
+  cleanForm = () => {
```

Comment:
this is not been used. You can remove it

Reply:
Removing it and the one in organization as well

Reply:
Removed this one and the organization version which was also not being called